ui/packages/catalog-ui-search/src/main/webapp/react-component/container/metacard-archive/metacard-archive.tsx

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+import * as React from 'react'
+import { hot } from 'react-hot-loader'
+import withListenTo, { WithBackboneProps } from '../backbone-container'
+import fetch from '../../utils/fetch'
+const announcement = require('component/announcement')
+const store = require('../../../js/store.js')
+const ResultUtils = require('../../../js/ResultUtils.js')
+const ConfirmationView = require('../../../component/confirmation/confirmation.view.js')
+import MetacardArchivePresentation from '../../presentation/metacard-archive'
+
+type Props = {
+  selectionInterface: any
+  model: any
+} & WithBackboneProps
+
+type State = {
+  collection: Backbone.Collection<Backbone.Model>
+  isDeleted: boolean
+  loading: boolean
+}
+
+class MetacardQuality extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+
+    const selectionInterface = props.selectionInterface || store
+    const collection = props.model || selectionInterface.getSelectedResults()
+
+    const isDeleted = collection.some((result: any) => {
+      result.isDeleted()
+    })
+
+    this.state = {
+      collection,
+      isDeleted,
+      loading: true,
+    }
+  }
+
+  componentDidMount() {
+    this.setState({ loading: false })
+  }
+
+  handleArchive = () => {
+    const body = JSON.stringify(
+      this.state.collection.map((result: any) => {
+        return result.get('metacard').get('id')
+      })
+    )
+
+    const self = this
```

Comment:
We shouldn't be using the `self = this` pattern in our code. You should pull 

```javascript
      async function(confirmation: any) {
```

into a named arrow function (possibly `onArchiveConfirm`), then it will auto bind `this`.